### DISCUSSION TITLE
Problem with Python Environment Name

### DISCUSSION THREAD

**Discussion by afigueroa1xp** [JanDeDobbeleer/oh-my-posh#5405] [Troubleshoot] [answered]
Hello 

I've customized one of the themes and added to the prompt the name of the virtual Python environment I'm working in, but I'm having a problem. In the template I have activated the name of the virtual environment to be shown if it is present:

![code](https://github.com/user-attachments/assets/00ce8af4-5032-46fb-a87c-978e63d97e7a)

But if my virtual environment is called "venv" which is the default name, I don't see that name, but the name of the project:

![Captura de pantalla 2024-08-02 131915-2](https://github.com/user-attachments/assets/5f9d4de5-cbc0-4ec1-a5f8-2d5a6709913d)

That does not happen if the name of the virtual environment is different from "venv", for example, you can see in the screenshot that when I have activated the Conda virtual environment, there is the correct name:

![Captura de pantalla 2024-08-02 131812](https://github.com/user-attachments/assets/5ce9211d-ae4c-4ba4-b39d-796ac82c038a)

What am I doing wrong?

**Comment by JanDeDobbeleer** [comment]
@afigueroa1973 you need to enable `display_default`. 

EDIT: wait, you did. I'll have a look.

**Comment by v-amorim** [comment, marked as answer]
When I was trying out the python segment, I found this out that might be related: 
If the `.venv/pyvenv.cfg` has the property `prompt = any name`, oh-my-posh uses that name when displaying the `{{ .Venv }}` property.

**Reply by v-amorim** [reply]
@afigueroa1973 check if this property is on yours and if removing it works as you intended

**Reply by afigueroa1xp** [reply]
When the default virtual environment is created with the `python -m venv venv` command, a prompt is not created in the `pyvenv.cfg` file. If I add the prompt line manually with the name I want to be displayed, now it is displayed. 

I guess from now on I'll have to create all my virtual environments using the `python -m venv venv --prompt="desire_name"` command.

Thanks for your answer @v-amorim
